RFC: USB: Use maximum-speed DTS parameters

finikorg opened the pull request:
Add better HS support

Fixes #11000

finikorg committed:
usb: usb_dc_sam: Fix maximum-speed handling

Fix handling for sam boards.
Replace definitions like DT_USBHS_MAXIMUM_SPEED with
DT_USB_MAXIMUM_SPEED which is more correct, since HS implies that
speed is High Speed and similar to definitions for other boards.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

aurel32 commented:
> At the moment maximum-speed is a string making it impossible to use
> with preprocessor macros.

The initial version of the PR didn't use strings. It has been switched to match the linux bindings. See:
https://github.com/zephyrproject-rtos/zephyr/pull/9062#pullrequestreview-146441447

carlescufi commented:
@aurel32 understood, thanks for that.
@erwango can you also weigh in here?

@finikorg @jfischer-phytec-iot I think it might be better to stay with strings given @galak's original comment

carlescufi commented on include/dt-bindings/usb/usb.h:
```diff
@@ -0,0 +1,15 @@
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_USB_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_USB_H_
+
+#define USB_MAXIMUM_SPEED_LOW_SPEED	0
```
I would move those to `usb_dc.h` or similar and then have some code that translates the string into a variable containing one of the values. I would do this directly in PR #14977

finikorg replied:
we need this kind of stuff 
b441bba68a64a3ba8a6f389b4845b8878d632921

finikorg replied:
And especially here 6098067af34951b0ef7bf2ab1dae2721213f49a1. This is just the example, should be changed to every place we define endpoints.

finikorg committed:
usb: Define MPS based on maximum speed parameter

Define MPS for endpoints based on maximum speed DTS parameter.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
tests: usb: Use USB_MAX_BULK_MPS in tests

Use defined USB_MAX_BULK_MPS for endpoint wMaxPacketSize.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

galak reviewed (dismissed):
The DTS proper maximum-speed should be a string, it should NOT be a int.  This matches dts binding specs from Linux for this property.

finikorg commented:
> The DTS proper maximum-speed should be a string, it should NOT be a int. This matches dts binding specs from Linux for this property.

How can we use strings for preprocessor macros?

finikorg commented:
Maybe our dts parser script can create out of strings like `DT_USB_MAXIMUM_SPEED = "high-speed"` something like
```
#define DT_USB_MAXIMUM_SPEED_HIGH_SPEED 1
```

jfischer-phytec-iot commented on include/usb/usb_device.h:
```diff
@@ -74,6 +75,17 @@ extern "C" {
 
 #define MAX_PACKET_SIZE0    64        /**< maximum packet size for EP 0 */
 
+#if defined(DT_USBHS_MAXIMUM_SPEED) && \
```
I think doing it this way is not good, readability suffers and worse, we need to touch it again to support more then one device controller, e.g. a SoC may have one FS controller and one HS controller. That why I want to check it [explicit ](https://github.com/zephyrproject-rtos/zephyr/pull/15147/commits/9657a10e2bedf655f5b27445ef39c8147e0442c8)for at least HS and FS.

finikorg replied:
To support several device controllers we need to rewrite a lot anyway, so it is probably better to touch less code. And we do compare against one global `DT_USBHS_MAXIMUM_SPEED` which does not depends on the controller. Probably that check should be done in `usb_dc_***` function.

finikorg replied:
3c750df3ddf6ba464389b49f69273a35abfbb118 make usb device controller check

carlescufi commented:
Will be reworked

finikorg committed:
drivers: usb: usb_dc_stm32: Use maximum-speed enum

Use maximum speed enum making it consistent with the rest of the code
and making code faster.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: dt-bindings: Change enum to macros

Use macros since we need to use them in the preprocessor conditionals.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: device: Return Device Qualifier for HS devices

For the devices defining DT_USB_MAXIMUM_SPEED_ENUM and equal to
DT_USB_MAXIMUM_SPEED_HIGH_SPEED return Device Qualifier.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
tests: usb: desc_sections: Use standard MPS and check size

Use USB_MAX_BULK_MPS instead of hardcoded TEST_BULK_EP_MPS and check
size for High Speed.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

carlescufi commented:
@jfischer-phytec-iot can you re-review this one?

galak commented:
@finikorg is this still relevant?  If not please close or update the PR.

finikorg commented:
Closing for now
